app/src/main/java/com/example/jetweatherforecast/screens/main/MainScreen.kt

```diff
@@ -45,26 +45,22 @@ fun MainScreen(
     if (!unitFromDb.isNullOrEmpty()) {
         unit = unitFromDb[0].unit.split(" ")[0].lowercase()
         isImperial = unit == "imperial"
-
-        val weatherData = produceState<DataOrException<Weather, Boolean, Exception>>(
-            initialValue = DataOrException(loading = true)) {
-            value = mainViewModel.getWeatherData(city = curCity,
-              units = unit)
-        }.value
-
-        if (weatherData.loading == true) {
-            CircularProgressIndicator()
-        }else if (weatherData.data != null) {
-            MainScaffold(weather = weatherData.data!!, navController,
-                isImperial = isImperial)
-
-        }
-
```

Comment:
Lo unico que hice fue mover este codigo afuera del `if (!unitFromDb.isNullOrEmpty()) `
El problema era que si no habia nada en el database (and there's nothing when the app first runs), the screen wasn't rendering.